src/app/pages/coursesList/coursesList.component.ts

```diff
@@ -21,34 +20,67 @@ export class CoursesListComponent implements OnInit, OnDestroy {
 	private modalTitle: string = `Delete this course?`;
 	private currId: number;
 
-	constructor(private courseService: CourseService, private loaderService: LoaderService) {
-		console.log('Courses page constructor');
+	constructor(
+		private courseService: CourseService,
+		private loaderService: LoaderService) {
 
 		this.courseList = [];
 	}
 
 	public ngOnInit() {
+		let params = this.getUrlParams();
 		this.loaderService.Show();
-		this.courseServiceSubscription = this.courseService.getCourses().subscribe((res: Course[]) => {
-			console.log('Got data', res);
-			this.courseList = res;
-			setTimeout(() => { this.loaderService.Hide(); }, 200);
+		this.courseServiceSubscription = this.courseService
+		.getCourses(params.start, params.count)
+		.subscribe((res: Course[]) => {
+			this.processCourses(res);
 		});
 	}
 
 	public ngOnDestroy() {
 		this.courseServiceSubscription.unsubscribe();
 	}
 
-	private getSearchResult(data) {
-		this.courseList = data;
+	private processCourses(res) {
+		this.courseList = res;
+		setTimeout(() => { this.loaderService.Hide(); }, 200);
+	}
+
+	private getSearchResult(searchString) {
+		let params = this.getUrlParams();
+		console.log('searchString is: ', searchString);
+		this.courseServiceSubscription = this.courseService
+		.getCourses(params.start, params.count, searchString)
+		.subscribe((res: Course[]) => {
+			this.processCourses(res);
+		});
 	}
 
 	private askForReals(id) {
 		this.currId = id;
 		this.modalHidden = false;
 	}
 
+	private getUrlParams() {
+		const regex = /[?&]([^=#]+)=([^&#]*)/g;
+		const url = window.location.href;
+		let params = {start: undefined, count: undefined};
+		let match;
+		while (match = regex.exec(url)) {
+			params[match[1]] = match[2];
+		}
+		return params;
+	}
+
+	private giveNextCouses() {
+		let params = this.getUrlParams();
+		this.courseServiceSubscription = this.courseService
+		.getCourses(parseInt(params.start, 10) + parseInt(params.count, 10), params.count)
+		.subscribe((res: Course[]) => {
+			this.processCourses(res);
+		});
+	}
+
```

Comment:
Please, fix indentation in and typo  `giveNextCouses()` and indentation in `getSearchResult()` methods